### PR TITLE
options: Do not use linked lists of pointers!

### PR DESCRIPTION
**Pull request by adamemerson**
it is almost certainly the case that you should not use linked lists in C++.

if you don't believe me, there's this Bjarne Stroustrup guy who, I've heard, knows a few things about C++. He also thinks you shouldn't use linked lists: https://youtube.com/watch?v=YQs6IC-vgmo

Especially if all we're doing is pushing things onto the end.

### CODE REVIEW

File: src/common/options.h [hunk 1]
```diff
@@ -233,7 +233,7 @@ struct Option {
     return *this;
   }
 
-  Option& set_enum_allowed(const std::list<std::string> allowed)
+  Option& set_enum_allowed(const std::vector<std::string> allowed)
```
Comment (cbodley) [inline src/common/options.h]: can pass by const reference here?

**Comment by mattbenjamin** [comment]
@adamemerson boost::intrusive::(s)list

**Comment by yuriw** [comment]
RBD
http://pulpito.ceph.com/yuriw-2017-09-27_15:26:37-rbd-wip-yuri-testing4-2017-09-27-1402-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-09-27_20:29:25-rbd-wip-yuri-testing4-2017-09-27-1402-distro-basic-smithi/

RADOS

http://pulpito.ceph.com/yuriw-2017-09-27_15:37:08-rados-wip-yuri-testing4-2017-09-27-1402-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-09-28_02:30:38-rados-wip-yuri-testing4-2017-09-27-1402-distro-basic-smithi/

File: src/common/options.h (hunk 1 above)
Comment (tchaikov) [reply src/common/options.h]: @cbodley #18034

**Pull request merged**